llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -1771,11 +1771,14 @@ Instruction *InstCombinerImpl::foldICmpAndConstConst(ICmpInst &Cmp,
       return new ICmpInst(NewPred, X, Zero);
     }
 
-    APInt NewC2 = *C2;
     KnownBits Know = computeKnownBits(And->getOperand(0), 0, And);
+    if (Know.One.intersects(*C2))
+      return replaceInstUsesWith(
+          Cmp, ConstantInt::getBool(Cmp.getType(), isICMP_NE));
```

Comment:
Is there another place we could hijack an existing knownbits calls?
This could be generalized to `!((Know.One & *C2) & ~C1).isZero()` Maybe put this in `foldICmpUsingKnownBits`?

Reply:
Before: http://llvm-compile-time-tracker.com/compare.php?from=a747e86caacf0ba3d5272474da8c10eb0b67dbc2&to=fd6783edc42218cf3d197cefe8b89cea41589f4d&stat=instructions%3Au

Reply:
Hmm? What are you trying to indicate?

Reply:
Fixed by converting `isKnownZero` to use SimplifyQuery.